lib/node/http.js

```diff
@@ -432,6 +434,133 @@ class HTTP extends Server {
 
       res.json(200, { success: true });
     });
+
+    this.get('/name/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      if (!name || !rules.verifyName(name))
+        throw new Error('Invalid parameter.');
+
+      const network = this.network;
+      const height = this.chain.height;
+      const nameHash = rules.hashName(name);
+      const reserved = rules.isReserved(nameHash, height + 1, network);
+      const [start, week] = rules.getRollout(nameHash, network);
+      const ns = await this.chain.db.getNameState(nameHash);
+
+      let info = null;
+
+      if (ns) {
+        if (!ns.isExpired(height, network))
+          info = ns.getJSON(height, network);
+      }
+
+      return res.json(200, {
+        start: {
+          reserved: reserved,
+          week: week,
+          start: start
+        },
+        info
+      });
+    });
+
+    this.get('/resource/hash/:hash', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const hash = valid.bhash('hash');
+
+      if (!hash)
+        throw new Error('Invalid hash.');
+
+      const ns = await this.chain.db.getNameState(hash);
+      const height = this.chain.tip.height;
+      const network = this.network;
+
+      if (!ns)
+        return res.json(404);
+
+      return res.json(200, ns.getJSON(height,network));
+    });
+
+    this.get('/resource/name/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      if (!name || !rules.verifyName(name))
+        throw new Error('Invalid name.');
+
+      const nameHash = rules.hashName(name);
+      const ns = await this.chain.db.getNameState(nameHash);
+
+      if (!ns || ns.data.length === 0)
+        return null;
+
+      const resource = Resource.decode(ns.data);
+
+      return res.json(200, resource.getJSON(name));
+    });
+
+    this.get('/proof/name/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      if (!name || !rules.verifyName(name))
+        throw new Error('Invalid name.');
+
+      const tip = this.chain.tip.hash;
+      const height = this.chain.tip.height;
+      const root = this.chain.tip.treeRoot;
+      const nameHash = rules.hashName(name);
+      const proof = await this.chain.db.prove(root, nameHash);
+
+      return res.json(200, {
+        hash: tip.toString('hex'),
+        height: height,
+        root: root.toString('hex'),
+        name: name,
+        key: nameHash.toString('hex'),
+        ...proof.toJSON()
```

Comment:
I don't love this solution, I think it would be more ideal to pass some data into `proof.toJSON` but then that would couple the proof to a particular blockchain implementation when it could be agnostic. The proof is defined in the `urkel` library, not `hsd`.

Maybe something like:

```
{
  proof: proof.toJSON(),
  chain: {
    height: height
    root: root.toString('hex')
    name: name
  }
```

We want to be able to support passing in an arbitrary `height` (create proofs at a certain height) eventually, that feels like it could be its own endpoint `GET /proof/name/:name/height/:height` or by adding a query param `GET /proof/name/:name?height=height`

Pinging @boymanjor @pinheadmz for opinions

Reply:
I don't want to opinionate too much since I'm new to this thread but... To compare, `rpc gettxoutproof` just returns a raw hex string, and has a matching verifier `rpc verifytxoutproof`. I think getting a proof is useless (?) without a matching endpoint to verify it, and that should perhaps dictate the output format of the `get proof` endpoint? 

Briefly checking this:

https://github.com/handshake-org/urkel/blob/c29466104b25f85ec2ef7a5025af625a6df21eda/lib/radix/proof.js#L256

`  verify(root, key, hash, bits) {...`

... I would expect a proof-getter to return the proof (either JSON or serialized hex) along with the current (by default) tree root and perhaps the `key`, which is just the name hash, but the user might not have the hash if they are using this endpoint.